lstchain/tools/lstchain_fit_intensity_scan.py

```diff
@@ -0,0 +1,364 @@
+                    fig.suptitle(f"{channel[chan]} channel", fontsize=25)
+                    ax=plt.subplot(1, 1, 1)
+                    ax.grid(True)
+                    ax.yaxis.set_major_locator(MaxNLocator(integer=True))
+                    ax.xaxis.set_major_locator(MaxNLocator(integer=True))
+                    ax.yaxis.set_major_locator(plt.MultipleLocator(1))
+
+                    plt.plot(np.median(self.signal[chan],axis=0),self.selected_runs[chan],"o")
+                    plt.xlabel(r'$\mathrm{\overline{Q}-\overline{ped}}$ [ADC]')
+                    plt.ylabel(r'Runs used in the fit')
+
+                    pdf.savefig()
+
+                    # plot the fit results and residuals for four arbitrary  pixels
+                    fig = plt.figure((chan+1)*10, figsize=(11, 22))
+                    fig.suptitle(f"{channel[chan]} channel", fontsize=25)
+
+                    pad = 0
+                    for pix in [0,600,1200,1800]:
+                        pad+=1
+                        plt.subplot(4, 2, pad)
+                        plt.grid(which='minor')
+
+                        mask = self.unusable_pixels[chan][pix]
+                        sig = np.ma.array(self.signal[chan][pix], mask=mask).compressed()
+                        var = np.ma.array(self.variance[chan][pix], mask=mask).compressed()
+                        popt = self.fit_parameters[chan, pix]
+
+                        # plot points
+                        plt.plot(sig, var, 'o', color="C0")
+
+                        # plot fit
+                        min_x=min(1000,np.min(sig)*0.9)
+                        max_x=max(10000,np.max(sig)*1.1)
+                        x = np.arange(np.min(sig),np.max(sig))
+
+                        plt.plot(x, quadratic_fit(x, *popt), '--', color="C1",
+                                 label=f'Pixel {pix}:\ng={popt[0]:5.2f} [ADC/pe] , B={popt[1]:5.3f}')
+                        plt.xlim(min_x, max_x)
+                        plt.xlabel(f'Q-ped [ADC]')
+                        plt.ylabel(r'$\mathrm{\sigma_Q^2-\sigma_{ped}^2}$ [$ADC^2$]')
+                        plt.xscale('log')
+                        plt.yscale('log')
+                        plt.legend()
+
+                        # plot residuals
+                        pad += 1
+                        plt.subplot(4,2,pad)
+                        plt.grid(which='both',axis='both')
+
+                        popt = self.fit_parameters[chan, pix]
+                        plt.plot(sig, (quadratic_fit(sig, *popt) - var) / var * 100, 'o', color="C0")
+                        plt.xlim(min_x,max_x)
+                        plt.xscale('log')
+                        plt.ylabel(f'fit residuals %')
+                        plt.xlabel(f'Q-ped [ADC]')
+                        plt.hlines(0, 0, np.max(sig), linestyle='dashed', color="black")
+
+                    plt.tight_layout()
+                    pdf.savefig()
+
+
+
+def quadratic_fit(t, b=1, c=1):
+    f2 = 1.222
```

Comment:
it is not nice to have this F2 value hardcoded here (I think it only affects the gain, while you want to compute first of all the quadratic term, but still, the gain itself is also put in the output, so the value should be consistent with what is used everywhere else)

Reply:
yes, my problem was how to pass a constant parameter to the fitted function. I found a solution